full-service/src/service/wallet.rs

```diff
@@ -190,6 +214,15 @@ pub enum JsonCommandResponse {
     },
 }
 
+// Helper method to escape quotes for json responses.
+fn format_error(e: WalletServiceError) -> String {
+    format!(
```

Comment:
Is it not possible to use the json! macro to do this properly? This looks like a problematic hack.
https://github.com/serde-rs/json#constructing-json-values

Reply:
I ran into issues deriving `serde::Serialize` for WalletServiceError because several of the wrapped errors from external crates don't implement `serde::Serialize`. I will wrap it in `json!().to_string()` to make it clear that it's json, but will still need to do the annoying quote replacement hack. The reason for returning both the stringified `error` and the details from `error.to_string()` is to make sure to get the full formatted displaydoc output.

Reply:
Actually, the json macro handles the quotes!